cloudwatch-logs-s3-forwarder/main.tf

```diff
@@ -0,0 +1,112 @@
+resource "random_string" "unique_identfier" {
+  length  = 8
+  special = false
+}
+
+resource "aws_iam_role" "lambda_role" {
+  name = "lambda_role_${random_string.unique_identfier.result}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": "stmtLambdaAssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "allow_lambda_to_log" {
+  name        = "cw_for_lambda_${random_string.unique_identfier.result}"
+  description = "Lets the Lambda function write to cloudwatch"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams"
+    ],
+      "Resource": [
+        "arn:aws:logs:*:*:*"
+    ]
+  }
+ ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "allow_s3_puts" {
+  name        = "s3_for_lambda_${random_string.unique_identfier.result}"
+  description = "Lets the Lambda function write to the bucket"
+
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": [
+        "s3:PutObject",
+        "s3:PutObjectAcl"
+      ],
+      "Effect": "Allow",
+      "Resource": ["arn:aws:s3:::${var.log_bucket_name}/*"]
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_cw_attach" {
+  role       = "${aws_iam_role.lambda_role.name}"
+  policy_arn = "${aws_iam_policy.allow_lambda_to_log.arn}"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_s3_attach" {
+  role       = "${aws_iam_role.lambda_role.name}"
+  policy_arn = "${aws_iam_policy.allow_s3_puts.arn}"
+}
+
+resource "aws_lambda_function" "bucket_forwarder" {
+  s3_bucket     = "${var.lambda_bucket}"
+  s3_key        = "${var.lambda_artifact_s3_key}"
+  function_name = "${var.lambda_function_name}"
+  role          = "${aws_iam_role.lambda_role.arn}"
+  handler       = "com.telia.aws.cloudwatchtoremotebucket.Handler::handleRequest"
```

Comment:
This contains telia specific code and is hardcoded, but should not.

Reply:
I added both handler and runtime to variables, Now the module is super generic and even more usable. Nice! I initially thought that this module only was appropriate to use with our Lambda.